### ISSUE TITLE
use dlprof tool to analyse nvtx-keras-script

### ISSUE THREAD

**Issue by fenfaqingnian** [NVIDIA/nvtx-plugins#20]
The dlprof tool analyzed the deep model and proposed that the data shape did not meet the requirements of tensor core. The original script set five full connection layers, namely, shape (8,1024), (1024,1024), (1024,512), (512,1) and batch=128. When the batch=64, the improvements proposed by DLprof were resolved. why？I did not change the shape(512,1) to (512,8)

this is my code in github, https://github.com/fenfaqingnian/dlprof_v100/tree/master/Profiler_DLprof_TF1-master

**Comment by tgerdesnv** [comment]
What container are you running in?

The DLProf recommendation is stale and is going to be removed in future versions.  It is no longer the case that a bad shape will prevent you from using tensor cores. However, it is still recommended to use multiples of 8 when possible to avoid the overhead of the cuda library padding the shape for you.

**Comment by fenfaqingnian** [comment]
container：Tensorflow 1.x: nvcr.io/nvidia/tensorflow:<xx.yy>-tf1-py3
GPU：V100
I have a confusing point, how to use tensor Core , which must meet the channels that multiples of 8 and data type must meet the float16,Can't we have one without the other？
I have a model,textcnn,I specify the environment variable to use AMP and XLA,but the model can't use tensor core ,why?

**Comment by tgerdesnv** [comment]
You can learn a lot about how to optimize performance here:  https://docs.nvidia.com/deeplearning/performance/index.html#tensor-layout
Fully connected layers: https://docs.nvidia.com/deeplearning/performance/dl-performance-fully-connected/index.html
Tensor core requirements: https://docs.nvidia.com/deeplearning/performance/dl-performance-matrix-multiplication/index.html#requirements-tc


For V100, FP16 datatype is required to use tensor cores.  Using AMP should provide that for you.

What are the <xx.yy> in your above container?  There was a known bug in DLProf where tensor core usage wasn't correctly identified for some kernels until 20.08 (which was very recently released).  If you didn't run on 20.08 please try that.

**Comment by fenfaqingnian** [comment]
https://docs.nvidia.com/deeplearning/frameworks/tensorflow-release-notes/rel_20-07.html#rel_20-07
this is the container that I use.
thanks for your answer,I will  Continue to explore

**Comment by DEKHTIARJonathan** [comment]
if we answered your questions. Please close the issue :)
